Source/WTF/wtf/CompletionHandler.h

```diff
@@ -33,7 +33,12 @@
 namespace WTF {
 
 template<typename> class CompletionHandler;
-enum class CompletionHandlerCallThread { MainThread, ConstructionThread };
+class CompletionHandlerCallThread {
+public:
+    static inline constexpr auto ConstructionThread = currentThreadLike;
+    static inline constexpr auto MainThread = mainThreadLike;
+    static inline constexpr auto AnyThread = anyThreadLike;
```

Comment:
As I said offline, I understand you have a particular need for this coming up. However, I hope usage of this doesn't get expanded without a good reason because I don't think this is a good pattern in general.